src/main/java/org/folio/service/mapping/converter/RecordConverter.java

```diff
@@ -1,62 +1,94 @@
-           String holdingId = holding.getString("id");
-           List<JsonObject> currentItems = items.stream()
-             .filter(item -> holdingId.equals(item.getString("holdingsRecordId")))
-             .collect(Collectors.toList());
-           holding.put("items", currentItems);
-         }
-       }
-       holdings.forEach(holding -> holding.put("instanceHrId", instanceHrId));
-       appendHoldingsItems.put("holdings", new JsonArray(holdings));
-     }
-   }
+  protected void fetchHoldingsAndItems(MappingProfile mappingProfile, OkapiConnectionParams params, String recordId,
+                                       String recordHrId, RecordType recordType, JsonObject appendHoldingsItems, String jobExecutionId) {
+    if (isTransformationRequired(mappingProfile)) {
+      LOGGER.debug("Fetching holdings/items for instance");
+      List<JsonObject> holdings;
+      if (recordType.equals(RecordType.INSTANCE)) {
+        holdings = recordLoaderService.getHoldingsForInstance(recordId, jobExecutionId, params);
+      } else {
+        LoadResult holding = recordLoaderService.getHoldingsById(Collections.singletonList(recordId), jobExecutionId, params);
+        holdings = holding.getEntities();
+      }
+      if (mappingProfile.getRecordTypes().contains(RecordType.ITEM) && CollectionUtils.isNotEmpty(holdings)) {
+        List<String> holdingIds = holdings.stream().map(holding -> holding.getString("id")).collect(Collectors.toList());
+        List<JsonObject> items = recordLoaderService.getAllItemsForHolding(holdingIds, jobExecutionId, params);
+        for (JsonObject holding : holdings) {
+          String holdingId = holding.getString("id");
+          List<JsonObject> currentItems = items.stream()
+            .filter(item -> holdingId.equals(item.getString("holdingsRecordId")))
+            .collect(Collectors.toList());
+          holding.put("items", currentItems);
+        }
+      }
+      if (recordType.equals(RecordType.INSTANCE)) {
+        holdings.forEach(holding -> holding.put("instanceHrId", recordHrId));
+      }
+      appendHoldingsItems.put("holdings", new JsonArray(holdings));
+    }
+  }
+
+  protected LoadResult fetchInstancesForHoldings(List<JsonObject> holdings, String jobExecutionId, OkapiConnectionParams params) {
+    List<String> instanceIds = getInstanceIdsFromHoldings(holdings);
+    return recordLoaderService.loadInventoryInstancesBlocking(instanceIds, jobExecutionId, params, INVENTORY_LOAD_PARTITION_SIZE);
+  }
+
+  private List<String> getInstanceIdsFromHoldings(List<JsonObject> holdings) {
+    return holdings.stream()
+      .map(JsonObject.class::cast)
+      .map(json -> json.getString("instanceId"))
+      .collect(Collectors.toList());
+  }
+
+  /**
+   * Fetches items if Transformations specify Record type "ITEM"
+   */
+  protected void fetchItemsIfRequired(MappingProfile mappingProfile, OkapiConnectionParams params, String holdingUUID, JsonObject appendHoldingsItems, String jobExecutionId) {
```

Comment:
Items aren't needed for now but I left the code since it will be required when supporting custom mp will be implemented.